circuits/cpp/src/aztec3/circuits/kernel/private/utils.cpp

```diff
@@ -73,4 +76,19 @@ PreviousKernelData<NT> dummy_previous_kernel(bool real_vk_proof = false)
     return previous_kernel;
 }
 
-}  // namespace aztec3::circuits::kernel::private_kernel::utils
\ No newline at end of file
+NT::address compute_ethereum_address_from_public_key(const NT::secp256k1_point& public_key)
+{
+    const std::vector<uint8_t> x_bytes = public_key.x.to_buffer();
+    const std::vector<uint8_t> y_bytes = public_key.y.to_buffer();
+    std::vector<uint8_t> public_key_bytes;
+    public_key_bytes.insert(public_key_bytes.end(), x_bytes.begin(), x_bytes.end());
+    public_key_bytes.insert(public_key_bytes.end(), y_bytes.begin(), y_bytes.end());
+    std::vector<uint8_t> public_key_hash = stdlib::keccak<UltraComposer>::hash_native(public_key_bytes);
+    std::vector<uint8_t> chopped_public_key_hash(public_key_hash.size(), 0);
+    for (size_t i = 12; i < 32; i++) {
+        chopped_public_key_hash[i] = public_key_hash[i];
+    }
```

Comment:
So the public key is all 0s except bytes 12 to 31 which are the public key hash? Can we link a doc here and/or add a comment?

Reply:
I've linked the reference and the section in it.